https://issues.redhat.com/browse/ACM-8319 (#5828)
open-cluster-management/rhacm-docs: business_continuity/backup_restore/backup_arch.adoc

    velero.io/exclude-from-backup: "true"
----

[#backup-files-created-by-a-schedule]
== Backup files created by an {product-title-short} schedule 

You can use an {product-title-short} schedule to backup hub resources, which are grouped in separate backup files, based on the resource type or label annotations.

> The `backupschedule.cluster.open-cluster-management.io` resource creates a set of four `schedule.velero.io` resources. These `schedule.velero.io` resources generate the backup files, which are also called resources. 

Reviewer: I see `BackupSchedule` on line 4
```suggestion
The `BackupSchedule.cluster.open-cluster-management.io` resource creates a set of four `schedule.velero.io` resources. These `schedule.velero.io` resources generate the backup files, which are also called resources. 
```

The scheduled backup files are `backup.velero.io`. See the following table to view the descriptions of these scheduled backup files: 

.Resource groups table
|===
| Scheduled backup | Description

| _Credentials backup_
| Stores the following: Hive credentials, {product-title-short}, user-created credentials, and `ConfigMaps`. The name of this backup file is `acm-credentials-schedule-<timestamp>`. 

| _Resources backup_
| Contains one backup for the {product-title-short} resources, `acm-resources-schedule-<timestamp> backup` and one for generic resources, `acm-resources-generic-schedule-<timestamp>`. Any resources annotated with the backup label, `cluster.open-cluster-management.io/backup`, are stored under the backup, `acm-resources-generic-schedule- backup`. The exceptions are `Secrets` or `ConfigMap` resources which are stored under the backup `acm-credentials-schedule-<timestamp>`. 

| _Managed clusters backup_
| Contains only resources that activate the managed cluster connection to the hub cluster, where the backup is restored. The name of this backup file is `acm-managed-clusters-schedule-<timestamp>`.
|===

> === Additional resources 

Reviewer: Additional resources should be at the end of the file

Reviewer: Recommending to remove this section to avoid broken links. We will revisit adding this after the team gets a clean smapshot for GA

For more information on backup files, see the following resources: 

> - Resources from the `acm-resources-generic-schedule` backup which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see xref:../business_continuity/backup_restore.adoc#restore-activation-resources[Restoring activation resources].  

Reviewer: ```suggestion
- Resources from the `acm-resources-generic-schedule` backup which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see xref:../business_continuity/backup_restore/backup_restore.adoc#restore-activation-resources[Restoring activation resources].  
```

> - Resources from the `acm-resources-generic-schedule- backup` which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-activation-resources[restoring activation resources].  

Reviewer: This link is hardcoded. I also saw that the "restoring activation resources" section is in the [Restoring a backup file](https://github.com/stolostron/rhacm-docs/blob/2.9_stage/business_continuity/backup_restore/backup_restore.adoc#restoring-activation-resources). Is "backup" a part of the name? 

I suggest that you use the following link syntax:

```suggestion
- Resources from the `acm-resources-generic-schedule` backup which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see xref:../business_continuity/backup_restore.adoc#restore-activation-resources[Restoring activation resources].  
```

> - Resources from the `acm-resources-generic-schedule` backup that are labeled with `cluster.open-cluster-management.io/backup:value` and where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see xref:../business_continuity/backup_restore.adoc#restore-passive-resources-check-backups[Restoring passive resources while checking for backups].  

Reviewer: ```suggestion
- Resources from the `acm-resources-generic-schedule` backup that are labeled with `cluster.open-cluster-management.io/backup:value` and where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see xref:../business_continuity/backup_restore/backup_restore.adoc#restore-passive-resources-check-backups[Restoring passive resources while checking for backups].  
```

> - Resources from the `acm-resources-generic-schedule` backup which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see xref:../business_continuity/backup_restore/backup_restore.adoc#restore-activation-resources[Restoring activation resources].  
> - Resources from the `acm-resources-generic-schedule` backup that are labeled with `cluster.open-cluster-management.io/backup:value` and where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see xref:../business_continuity/backup_restore/backup_restore.adoc#restore-passive-resources-check-backups[Restoring passive resources while checking for backups].  

Reviewer: Hopefully this helps
```suggestion
- Resources from the `acm-resources-generic-schedule` backup which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see link:../business_continuity/backup_restore/backup_restore.adoc#restore-activation-resources[Restoring activation resources].  
- Resources from the `acm-resources-generic-schedule` backup that are labeled with `cluster.open-cluster-management.io/backup:value` and where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see link:../business_continuity/backup_restore/backup_restore.adoc#restore-passive-resources-check-backups[Restoring passive resources while checking for backups].  
```

> - Resources from the `acm-resources-generic-schedule-` backup which are labeled with `cluster.open-cluster-management.io/backup:value` where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-passive-resources-check-backups[restoring passive resources]. 

Reviewer: Same thing, remove the hardcoded link. Based on the anchor, it seems like we should be pointing to the "Restoring passive resource while checking for backups" section.

```suggestion
- Resources from the `acm-resources-generic-schedule` backup that are labeled with `cluster.open-cluster-management.io/backup:value` and where the label value is not `cluster-activation`, are restored when the passive resources are restored. For more information on restoring passive resource, see xref:../business_continuity/backup_restore.adoc#restore-passive-resources-check-backups[Restoring passive resources while checking for backups].  
```

- `managedclustersetbinding.cluster.open-cluster-management.io`
- `clusterpool.hive.openshift.io`
- `clusterclaim.hive.openshift.io`
- `clustercurator.cluster.open-cluster-management.io`

[#backup-files-created-by-a-schedule]
== Backup files created by an {product-title-short} schedule 

> You can use an {product-title-short} schedule to generate backup files and can create a subsection to store and organize these backup files. 

Reviewer 2: this is confusing .. I am not sure what subsection this is referring to .. >> can create a subsection to store and organize these backup files. 

Let's take that out, the schedule only generates a set of velero backup files. It should say:

You can use an {product-title-short} schedule to backup hub resources, which are grouped in separate backup files, based on the resource type or label annotations.

The `backupschedule.cluster.open-cluster-management.io` resource creates a set of 4 `schedule.velero.io` resources. These `schedule.velero.io` resources generate the backup files. 

To view the list of scheduled backup files, run the following command: `oc get schedules -A | grep acm`. 

To view the descriptions of these scheduled backup files, see the following table: 

.Resource groups table
|===
| Resource | Description

| _Credentials backup_
| Stores the following: Hive credentials, {product-title-short}, user-created credentials, and `ConfigMaps`. The name of this backup file is `acm-credentials-schedule-<timestamp>`. 

| _Resources backup_

> | Contains one backup for the {product-title-short} resources, `acm-resources-schedule-<timestamp> backup` and one for generic resources, `acm-resources-generic-schedule-<timestamp>`. These resources use the backup label, `cluster.open-cluster-management.io/backup`.

Reviewer 2: Replace : These resources use the backup label, `cluster.open-cluster-management.io/backup`.

with : 
Any resource annotated with the backup label `cluster.open-cluster-management.io/backup` are stored under the acm-resources-generic-schedule-<timestamp> backup, with the exception of `Secrets` or `ConfigMap` resources which are placed under the `acm-credentials-schedule-<timestamp>` backup.

Resources from the acm-resources-generic-schedule-<timestamp> backup, labeled with cluster.open-cluster-management.io/backup: cluster-activation are being restored at cluster activation time. For more details on activation resources read https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-activation-resources

All other resources backup up by the acm-resources-generic-schedule-<timestamp> are restored when when passive resources are being restored. For more information on restoring passive resources read https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-passive-resources-check-backups.

Author: Awesome, thanks! Please let me know what you think about me moving these links and resources to an `additional resources` section. I think that reads better than including these links in the table. I wanted to keep the table concise. But of course, let me know how you think this layout and organization helps the users :)

| _Managed clusters backup_

> | Contains only resources that activate the managed cluster connection to the hub cluster, where the backup is restored. The name of this backup file is `acm-managed-clusters-schedule-<timestamp>`.

Reviewer 2: Add here : For more details on activation resources read https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-activation-resources

- `managedclustersetbinding.cluster.open-cluster-management.io`
- `clusterpool.hive.openshift.io`
- `clusterclaim.hive.openshift.io`
- `clustercurator.cluster.open-cluster-management.io`

[#backup-files-created-by-a-schedule]

> == Backup files created by an {product-title-short} schedule 

Reviewer: Should this be moved to line 53 to come before the _Resources restored at managed clusters activation time_ topic?

You can use an {product-title-short} schedule to backup hub resources, which are grouped in separate backup files, based on the resource type or label annotations.

> The `backupschedule.cluster.open-cluster-management.io` resource creates a set of 4 `schedule.velero.io` resources. These `schedule.velero.io` resources generate the backup files. 

Reviewer: ```suggestion
The `backupschedule.cluster.open-cluster-management.io` resource creates a set of four `schedule.velero.io` resources. These `schedule.velero.io` resources generate the backup files. 
```

To view the list of scheduled backup files, run the following command: `oc get schedules -A | grep acm`. 

> To view the descriptions of these scheduled backup files, see the following table: 

Reviewer: ```suggestion
See the following table to view the descriptions of these scheduled backup files: 
```

.Resource groups table
|===
| Resource | Description

> | _Credentials backup_

Reviewer: What's the `kind` for the resource? After you receive that info, also monospace it

Reviewer 2: The scheduled backup file ( the resources ) are backup.velero.io

Author: I added this info before the table. Let me know if you wanted a whole column in the table for `kind`

Reviewer: I imagined a camel-case value for the resource `kind`. It seems like `BackupSchedule` is the `kind`. What are yall thoughts about renaming the Resource column to Scheduled backups

| _Managed clusters backup_
| Contains only resources that activate the managed cluster connection to the hub cluster, where the backup is restored. The name of this backup file is `acm-managed-clusters-schedule-<timestamp>`.
|===

=== Additional resources 

For more information on backup files, see the following resources: 

- Resources from the `acm-resources-generic-schedule- backup` which are labeled with `cluster.open-cluster-management.io/backup:cluster-activation` are being restored at cluster activation time. For more information on activation resources, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-activation-resources[restoring activation resources].  

> - Resources with a backup by the `acm-resources-generic-schedule` are restored when the passive resources are restored. For more information on restoring passive resource, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/business_continuity/business-cont-overview#restore-passive-resources-check-backups[restoring passive resources]. 

Reviewer 2: looks great, one more change 

This : Resources with a backup by the `acm-resources-generic-schedule` are restored when the passive resources are restored.

Should read : 
Resources from the `acm-resources-generic-schedule- backup` which are labeled with `cluster.open-cluster-management.io/backup:value` where the label value is not `cluster-activation`, are restored when the passive resources are restored.

- `managedclustersetbinding.cluster.open-cluster-management.io`
- `clusterpool.hive.openshift.io`
- `clusterclaim.hive.openshift.io`

> - `clustercurator.cluster.open-cluster-management.i

Reviewer: ```suggestion
- `clustercurator.cluster.open-cluster-management.io`
```

- `managedclustersetbinding.cluster.open-cluster-management.io`
- `clusterpool.hive.openshift.io`
- `clusterclaim.hive.openshift.io`

> - `clustercurator.cluster.open-cluster-management.i

Reviewer: ```suggestion
- `clustercurator.cluster.open-cluster-management.io`
```
